managers/bosh-manager/BOSHManager.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const Promise = require('bluebird');
+const eventmesh = require('../../data-access-layer/eventmesh');
+const logger = require('../../common/logger');
+const CONST = require('../../common/constants');
+const BaseManager = require('../BaseManager');
+const DirectorService = require('./DirectorService');
+const errors = require('../../common/errors');
+const utils = require('../../common/utils');
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+
+class BOSHManager extends BaseManager {
+
+  init() {
+    const queryString = `state in (${CONST.APISERVER.RESOURCE_STATE.IN_QUEUE},${CONST.APISERVER.RESOURCE_STATE.UPDATE},${CONST.APISERVER.RESOURCE_STATE.DELETE})`;
+    return this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT, CONST.APISERVER.RESOURCE_TYPES.DIRECTOR)
+      .then(() => this.registerWatcher(CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT, CONST.APISERVER.RESOURCE_TYPES.DIRECTOR, queryString));
+  }
+
+  processRequest(requestObjectBody) {
+    return Promise.try(() => {
+        if (requestObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.IN_QUEUE) {
+          return this._processCreate(requestObjectBody);
+        } else if (requestObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.UPDATE) {
+          return this._processUpdate(requestObjectBody);
+        } else if (requestObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.DELETE) {
+          return this._processDelete(requestObjectBody);
+        }
+      })
+      .catch(err => {
+        logger.error('Error occurred in processRequest', err);
+        return eventmesh.apiServerClient.updateResource({
+          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+          resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+          resourceId: requestObjectBody.metadata.name,
+          status: {
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+            lastOperation: {
+              state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+              description: CONST.SERVICE_BROKER_ERR_MSG
+            },
+            error: utils.buildErrorJson(err)
+          }
+        });
+      });
+  }
+
+  _processCreate(changeObjectBody) {
```

Comment:
Can we rename variable **requestObjectBody** just to make consistent with caller [here](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/364/files#diff-58d6c650f9514c50affcb4394cc42210R24) ?

Reply:
Makes more sense to rename requestObjectBody to changeObjectBody as it is a change event from watch.